homeassistant/components/androidtv/media_player.py

```diff
@@ -83,6 +87,73 @@
     {vol.Required(ATTR_ENTITY_ID): cv.entity_ids, vol.Required(ATTR_COMMAND): cv.string}
 )
 
+# To be used for config validation
+VALID_PROPERTIES = ("audio_state", "media_session_state")
+VALID_CONDITIONS = VALID_PROPERTIES + ("wake_lock_size",)
+
+
+def valid_state_detection_rules(rules):
```

Comment:
Even better would be if we moved the whole validator to the library. We can do that by wrapping it in a factory function, that accepts a callback to be called when the validation fails. The callback should accept the message for the failed validation.

Then the user of the library can create a validator by calling the factory passing the callback. In our case the callback will raise `vol.Invalid`.

Reply:
Done.  I moved the state detection validation into the `androidtv` package.